write-fonts/src/graph/splitting.rs

```diff
@@ -1066,4 +1063,81 @@ mod tests {
             .sum();
         assert_eq!(total_c2recs, expected_n_c2_recs);
     }
+
+    #[test]
+    fn size_of_value_record_children_sanity() {
+        // let's have  single class1class, and three class2 classes
+        // we want a duplicate varidx, a null varidx, and a device table?
+
+        fn make_val_record(
+            x_advance: i16,
+            x_advance_device: impl Into<Option<wlayout::DeviceOrVariationIndex>>,
+        ) -> ValueRecord {
+            let format = ValueFormat::X_ADVANCE | ValueFormat::X_ADVANCE_DEVICE;
+            let rec = ValueRecord::new()
+                .with_explicit_value_format(format)
+                .with_x_advance(x_advance);
+            match x_advance_device.into() {
+                Some(dev) => rec.with_x_advance_device(dev),
+                None => rec,
+            }
+        }
+
+        let class_def1 = make_class_def(1, 4, 1);
+        let class_def2 = make_class_def(3, 3, 1);
+        let coverage = class_def1.iter().map(|(gid, _)| gid).collect();
+        let actual_device_table = Device::new(12, 15, &[118, 119, 127, 99]);
+        // sanity check the size of the device table, these are weird:
+        assert_eq!(crate::dump_table(&actual_device_table).unwrap().len(), 10);
+        let class1_records = vec![Class1Record::new(vec![
+            Class2Record::new(
+                make_val_record(5, None),
+                make_val_record(6, DeviceOrVariationIndex::variation_index(4, 20)),
+            ),
+            Class2Record::new(
+                make_val_record(7, DeviceOrVariationIndex::Device(actual_device_table)),
+                // a duplicate table
+                make_val_record(8, DeviceOrVariationIndex::variation_index(4, 20)),
+            ),
+            Class2Record::new(
+                make_val_record(9, DeviceOrVariationIndex::variation_index(6, 9)),
+                make_val_record(10, None),
+            ),
+        ])];
+        let ppf2 = PairPos::format_2(coverage, class_def1, class_def2, class1_records);
+
+        // now we need to pretend we're in the split_pair_pos_format_2 fn
+        let mut graph = TableWriter::make_graph(&ppf2);
+        assert!(graph.pack_objects());
+        let root_id = graph.root;
+        let ppf2_data = &graph.objects[&root_id];
+        let ppf2 = ppf2_data.reparse::<rgpos::PairPosFormat2>().unwrap();
+        assert_eq!(ppf2.class1_records().len(), 1);
+        let c1rec = ppf2.class1_records().get(0).unwrap();
+        let mut visited = HashSet::new();
+        let mut next_device_offset = 3;
+        assert_eq!(c1rec.class2_records.len(), 3);
+
+        // a little helper so we don't have to have this huge fn call in each assert
+        let mut test_closure = |record: &rgpos::ValueRecord| -> usize {
```

Comment:
"test_closure" is perhaps nto the most descriptive name, size_of_children would make the tests read much more clearly